src/Style.hs

```diff
@@ -300,130 +314,145 @@ initSpec = StySpec { spType = C.PointT,
     where
-        patternNamePairs :: [(Pattern, String)]
+        -- patternNamePairs :: [(Pattern, String)]
         patternNamePairs = zip (selPatterns sel) (spArgs spec)
-
-        add :: VarMap -> (Pattern, Name) -> VarMap
+        -- add :: VarMap -> (Pattern, Name) -> VarMap
         add dict (patt, name) = case patt of
             RawID _    -> dict
             WildCard wc -> M.insert wc name dict
 
--- | Returns all StySpecs in the StyDict that match a selector.
-matchingSpecs :: StyDict -> Selector -> [StySpec]
-matchingSpecs dict s = M.elems $ M.filter (matchSel s) dict
-
--- | Returns true if an object (declaration or constraint) matches the selector. A match is made when
--- * The types match, (and?)
--- * The number of arguments match, (and?)
--- * Identifier and arguments match. A wildcard matches with anything
-matchSel :: Selector -> StySpec -> Bool
-matchSel sel spec = all test (zip args patterns) &&
-                selTyp sel == spType spec &&
-                length args == length patterns
-    where
-        patterns = selPatterns sel
-        args = spArgs spec
-        -- dummies = selIds sel
-        test (a, p) = case p of
-            RawID i -> a == i
-            WildCard _ -> True
-
--- |  'procBlock' is called by 'getDictAndFns'. 'getDictAndFns' would fold this function on a list of blocks, a.k.a. a Style program, and accumulate objective/constraint functions, and a dictionary of geometries to be rendered.
--- TODO refactor procBlock and factor out helper functions
-procBlock :: (Autofloat a) =>
-    (StyDict, [(ObjFnOn a, Weight a, [Name], [a])], [(ConstrFnOn a, Weight a, [Name], [a])])
-    -> Block
-    -> (StyDict, [(ObjFnOn a, Weight a, [Name], [a])], [(ConstrFnOn a, Weight a, [Name], [a])])
-procBlock (dict, objFns, constrFns) (selectors, stmts) =
-          (newDict, objFns ++ newObjFns, constrFns ++ newConstrFns)
-    where -- Organized from used first to last
-        allVars :: M.Map Name Name
-        allVars = M.fromList $ zip k k where k = M.keys dict
 
-        -- Give specs (objects) in dict that are selected by the block's selectors
-        selectedSpecs :: [[(VarMap, StySpec)]]
-        selectedSpecs = map
-            (\sel -> let styspecs = matchingSpecs dict sel -- defined externally above procblock
-                         varmaps = map (getVarMap sel) styspecs in zip varmaps styspecs) selectors
+-- Gather a list of objective or constraint functions by mapping over mergedMaps
+-- genFns :: (VarMap -> [t] -> Stmt -> [t]) -> VarMap -> [t]
+genFns f stmts varmap = foldl (f varmap) [] stmts
 
+-- | 'procBlock' is called by 'getDictAndFns'. 'getDictAndFns' would fold this function on a list of blocks, a.k.a. a Style program, and accumulate objective/constraint functions, and a dictionary of geometries to be rendered.
+-- TODO: first split `stmts` into assigns, objs, and contrs and call procConstrFn and procObjFn with `map`?
+procBlock :: (Autofloat a) =>
```

Comment:
what does "proc" mean? consider giving all the "proc" functions more descriptive names (e.g. "blockToIR")